File: SourceCode/Projects/open.fda.gov/ApiDalc/DataObjects/IconStyle.vb
Public Class IconStyle

    ''' <summary>
    ''' Anchor location of icon
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property anchor() As Array = {0.5, 0.5}

    ''' <summary>
    ''' Anchor X units
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property anchorXUnits As String = "fraction"

    ''' <summary>
    ''' Anchor Y units
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property anchorYUnits As String = "fraction"

    ''' <summary>
    ''' Opacity of icon
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property opacity As Double = 0.85D

    ''' <summary>
    ''' Image location
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property src As String = "img/mapIcon/{0}.png"

End Class

File: SourceCode/AspCoreVersion/src/ShopAware.Core/DataObjects/IconStyle.cs
namespace ShopAware.Core
{
    public class IconStyle
    {
        #region Member Variables

        /// <summary>
        ///     Anchor location of icon
        /// </summary>
        /// <value></value>
        /// <returns></returns>
        /// <remarks></remarks>
        /// <summary>
        ///     Anchor X units
        /// </summary>
        /// <value></value>
        /// <returns></returns>
        /// <remarks></remarks>
        private string _anchorXUnits = "fraction";

        /// <summary>
        ///     Anchor Y units
        /// </summary>
        /// <value></value>
        /// <returns></returns>
        /// <remarks></remarks>
        private string _anchorYUnits = "fraction";

        /// <summary>
        ///     Opacity of icon
        /// </summary>
        /// <value></value>
        /// <returns></returns>
        /// <remarks></remarks>
        private double _opacity = (double) 0.85M;

        /// <summary>
        ///     Image location
        /// </summary>
        /// <value></value>
        /// <returns></returns>
        /// <remarks></remarks>
        private string _src = "img/mapIcon/{0}.png";

        #endregion

        #region Properties

        public string anchorXUnits
        {
            get
            {
                return _anchorXUnits;
            }
            set
            {
                _anchorXUnits = value;
            }
        }

        public string anchorYUnits
        {
            get
            {
                return _anchorYUnits;
            }
            set
            {
                _anchorYUnits = value;
            }
        }

        public double opacity
        {
            get
            {
                return _opacity;
            }
            set
            {
                _opacity = value;
            }
        }

        public string src
        {
            get
            {
                return _src;
            }
            set
            {
                _src = value;
            }
        }

        #endregion
    }
}